patchwork-extensions-blockentity/src/main/java/net/patchworkmc/mixin/extensions/blockentity/MixinWorld.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2020, 2019-2020
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.patchworkmc.mixin.extensions.blockentity;
+
+import java.util.Iterator;
+import java.util.List;
+
+import org.objectweb.asm.Opcodes;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.At.Shift;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.ModifyVariable;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+import net.minecraftforge.common.extensions.IForgeTileEntity;
+
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.util.Tickable;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.profiler.Profiler;
+import net.minecraft.world.World;
+import net.minecraft.world.chunk.WorldChunk;
+
+/**
+ * This Mixin implements {@link IForgeTileEntity#onLoad()} and {@link IForgeTileEntity#onChunkUnloaded()}.
+ */
+@Mixin(World.class)
+public class MixinWorld {
+	/////////////////////////////////////
+	/// addBlockEntity()
+	/////////////////////////////////////
+	@Inject(method = "addBlockEntity", at = @At("HEAD"))
+	private void onAddBlockEntity(BlockEntity blockEntity, CallbackInfoReturnable<Boolean> ci) {
+		World me = (World) (Object) this;
+
+		if (blockEntity.getWorld() != me) {
+			// Forge - set the world early as vanilla doesn't set it until next tick
+			blockEntity.setWorld(me);
```

Comment:
could this cause incompatibilities in fabric mods?

Reply:
No, the world field will be set anyway. Fabric mods are made under the assumption that world is not available upon creation, so they will not attempt to access it that early. I think this modification should not cause trouble to fabric mods. But I'm sure that a lot of Forge mods need this.